flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java

```diff
@@ -85,28 +86,29 @@ public class TestFlinkIcebergSink {
   private final int parallelism;
   private final boolean partitioned;
 
-  @Parameterized.Parameters(name = "format={0}, parallelism = {1}, partitioned = {2}")
+  @Parameterized.Parameters(name = "format={0}, parallelism = {1}, partitioned = {2}, branch = {3}")
```

Comment:
I have a little bit concern of the multiplication factor of Parameterized tests. I feel we don't need to test different file format, writer parallelism, partitioned or not for the branch write.

I am wondering if we should have a separate `TestFlinkIcebergSinkBranching`. it can be parameterized to test both v1 and v2 table format. also we can add more accurate assertion that write happened on the `test_branch` and verified that no snapshots/writes added to the `main` branch.

Reply:
Agreed, even I was thinking along the same lines because the file format/parallelism/partitioned are all independent of branch writes and we'd be adding a lot of unnecessary test cases. I think it makes sense to separate the tests here specifically for branches!

Reply:
@stevenzwu @jackye1995  I added this separate class and moved common logic between the sink/sinkv2 tests to TestFlinkIcebergSinkBase. After implementing, now I'm not sure if it's worth refactoring the code. Even though with the previous approach there's 2x the combinations which was unnecessary, it was a less intrusive to the test code and easier to read. Also running locally with 2x the combinations was still quite quick. Let me know your thoughts on what you prefer!

Reply:
here are some stats from one local run (a powerful M1 macbook) 
* TestFlinkIcebergSink: 36s
* TestFlinkIcebergSinkV2: 77s
* TestIcebergFilesCommitter: 19s

If we double it, the total build time for those 3 classes will increase from 132s (2m) to 264s (4m). If we add another dimension/multiplication after this, it will increase from 4+ mins to 8-9 mins.

That was my concern.

Reply:
Sure, yeah for branch tests the total time increase would be 2x. In that case, I will keep the current implementation which removes the extra dimension and separates the branch specific tests. let me know your thoughts @stevenzwu !